spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/integration/inbound/GcsStreamingMessageSource.java

```diff
@@ -35,14 +36,19 @@
  * @author João André Martins
  * @author Mike Eltsufin
  * @author Chengyuan Zhao
+ * @author Lukas Gemela
  */
 public class GcsStreamingMessageSource extends AbstractRemoteFileStreamingMessageSource<BlobInfo> {
 
-	public GcsStreamingMessageSource(RemoteFileTemplate<BlobInfo> template) {
-		super(template, null);
+	public GcsStreamingMessageSource(RemoteFileTemplate<BlobInfo> template, Comparator<BlobInfo> comparator) {
+		super(template, comparator);
 		doSetFilter(new GcsPersistentAcceptOnceFileListFilter(new SimpleMetadataStore(), "gcsStreamingMessageSource"));
 	}
 
+	public GcsStreamingMessageSource(RemoteFileTemplate<BlobInfo> template) {
```

Comment:
For smooth code flow this ctor has to be first. Because exactly this one calls another.

Reply:
updated accordingly